src/main/java/net/mcreator/element/types/ItemExtension.java

```diff
@@ -20,22 +21,32 @@
 
 import net.mcreator.element.GeneratableElement;
 import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.element.parts.NumberProcedure;
+import net.mcreator.element.parts.Procedure;
 import net.mcreator.minecraft.MinecraftImageGenerator;
 import net.mcreator.workspace.elements.ModElement;
 
 import java.awt.image.BufferedImage;
 
-public class Fuel extends GeneratableElement {
+public class ItemExtension extends GeneratableElement {
 
-	public int power;
-	public MItemBlock block;
+	public MItemBlock item;
 
-	public Fuel(ModElement element) {
+	public boolean enableFuel;
+	public NumberProcedure fuelPower;
+	public Procedure fuelSuccessCondition;
+
+	public boolean hasDispenseBehavior;
+	public Procedure dispenseSuccessCondition;
+	public Procedure dispenseResultItemstack;
+
+	public double layerChance;
+
+	public ItemExtension(ModElement element) {
 		super(element);
 	}
 
 	@Override public BufferedImage generateModElementPicture() {
-		return MinecraftImageGenerator.Preview.generateFuelPreviewPicture(getModElement().getWorkspace(), block);
+		return MinecraftImageGenerator.Preview.generateFuelPreviewPicture(getModElement().getWorkspace(), item);

```

Comment:
generateFuelPreviewPicture? Don't think this is relavant, so cleanup this code

Reply:
I kept this preview because I didn't know what could represent the entire element. I'm open to ideas.

Reply:
We can use mod element icon, no need to generate preview picture

Reply:
You don't have to override for default icon, as this is already handled in parent class

Reply:
Nothing is overridden...

Reply:
`@Override public BufferedImage generateModElementPicture() {`

to me it seems like this is overriding default generateModElementPicture

Reply:
![image](https://user-images.githubusercontent.com/38427877/169320187-141a018b-fe01-4aa3-9d49-f8fd70a9cbdc.png)
to me it seems like there is no such method

Reply:
Hmm, maybe github cached code view for me